docs/architecture/adr-023-protobuf-naming.md

```diff
@@ -0,0 +1,238 @@
+files as modules are the primary thing sub-packages are used for. Only items which
+are known to be used infrequently should have deep sub-package depths.
+
+For the Cosmos SDK, it is recommended that that we simply write `cosmos.bank`,
+`cosmos.gov`, etc. rather than `cosmos.x.bank`. In practice, most non-module
+types can go straight in the `cosmos` package or we can introduce a
+`cosmos.base` package if needed. Note that this naming _will not_ change
+go package names, i.e. the `cosmos.bank` protobuf package will still live in
+`x/bank`.
+
+### Message Naming
+
+#### Use simple verbs for transaction messages
+
+This is likely one of the most controversial recommendations here as it will
+change legacy go struct names.
+
+Historically all transaction messages in the SDK have been prefixed with `Msg`
+which is a shortening of msssage. To begin with "message", not being an action
+word, does not clearly indicate the concept of transaction or operation. So
+understanding that `Msg` indicates transaction likely requires knowledge of
+Cosmos SDK conventions. Once someone understands that `Msg` means transaction,
+however, it still is likely redundant information as the type name usually
+includes an action verb such as "send", "create" or "submit".
+
+Going forward, for both conciseness and clarity, transaction messages should
+simply use a descriptive action verb to indicate that this is a type which
+performs an action in a transaction. This should also be made clear in the
+documentation.
+
+To maintain compatibility with existing names in go code, an alias can be
+introduced for legacy type names.
+
+#### Use simple nouns for state types
+
+Nouns should be used for other types which are used in state, queries and as
+arguments to some transaction messages. Ex. `Coin`, `Proposal` and `Delegation`.
+
+### Service and RPC Naming
+
+[ADR 021](adr-021-protobuf-query-encoding.md) specifies that modules should
+implement a gRPC query service. We should consider the principle of conciseness
+for query service and RPC names as these may be called from persistent script
+modules such as CosmWasm. Also, users may use these query paths from tools like
+[gRPCurl](https://github.com/fullstorydev/grpcurl). As an example, we can shorten
+`/cosmos_sdk.x.bank.v1.QueryService/QueryBalance` to
+`/cosmos.bank.Query/Balance` without losing much useful information.
+
+#### Use just `Query` for the query service
+
+Instead of [Buf's default service suffix recommendation](https://github.com/cosmos/cosmos-sdk/pull/6033),
+we should simply use the shorter `Query` for query services.
+
+For other types of gRPC services, we should consider sticking with Buf's
+default recommendation.
+
+#### Omit `Get` and `Query` from query service RPC names
+
+`Get` and `Query` should be omitted from `Query` service names because they are
+redundant in the fully-qualified name. For instance, `/cosmos.bank.Query/QueryBalance`
+just says `Query` twice without any new information.
```

Comment:
@aaronc Do you recommendations for this? Or just simply omit all `Get` and `Query` prefixes.